db/channel_cache.go

```diff
@@ -335,8 +336,11 @@ func (c *channelCache) GetChanges(options ChangesOptions) ([]*LogEntry, error) {
 
 	// Nope, we're going to have to backfill from the view.
 	//** First acquire the _view_ lock (not the regular lock!)
+	// Track pending queries via StatKeyChannelCachePendingQueries expvar
+	c.context.DbStats.StatsCache().Add(base.StatKeyChannelCachePendingQueries, 1)
 	c.viewLock.Lock()
 	defer c.viewLock.Unlock()
+	c.context.DbStats.StatsCache().Add(base.StatKeyChannelCachePendingQueries, -1)
```

Comment:
Is this stat tracking how many queries are sat waiting to acquire the lock? Does it exclude the one currently holding the lock?

I was unsure if there was a defer missing here or not, but if the stat is exclusive to the current query I think it's fine.